train.py

```diff
@@ -365,15 +367,17 @@ def train_loop_fn(model, loader, device, context):
             tracker.add(sum(sample['nsentences'] for sample in samples))
         return tracker
 
-    def valid_loop_fn(model, loader, device, context):
-        trainer = trainers[str(device)]
+    def valid_loop_fn(device, trainer, loader):
         # reset validation loss meters
         for k in ['valid_loss', 'valid_nll_loss']:
             meter = trainer.get_meter(k)
             if meter is not None:
                 meter.reset()
         extra_meters = collections.defaultdict(lambda: AverageMeter())
         for i, sample in enumerate(loader):
+            if i+1 == len(loader._loader._loader):
+                #FIXME: last batches are incomplete
```

Comment:
Meant to be here?

Reply:
same, waiting for resolution on that one.